src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -10,16 +10,33 @@ public class ReadWriteList<E> {
     private ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
-        // write your code here
+        lock.writeLock().lock();
+        try {
+            list.add(element);
+        } finally {
+            lock.writeLock().unlock();
+        }
     }
 
     public E get(int index) {
-        // write your code here
-        return null;
+        lock.readLock().lock();
+        try {
+            if (index < 0 || index >= list.size()) {
+                throw new RuntimeException(
+                        "Index " + index + " out of bounds from size: " + list.size());
+            }
```

Comment:
we don't need that, list already has this functionality

Reply:
fixed